src/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -1220,19 +1229,16 @@
    time :- pls/Timestamp]
   (let [time (to-timestamp time)
         time-field (if (= entity :reports) :start_time :producer_timestamp)]
-    (if-let [db-timestamp (timestamp-of-newest-record entity certname time-field)]
-      (.after db-timestamp time)
-      false)))
+    (some-> entity
+     (timestamp-of-newest-record certname time-field)
+     (.after time))))
```

Comment:
I think this changes the function semantics from returning a real bool to returning true or nil. Was that intended?